linera-execution/src/system.rs

```diff
@@ -718,6 +685,81 @@ where
         Ok((outcome, new_application))
     }
 
+    pub async fn transfer(
+        &mut self,
+        authenticated_signer: Option<Owner>,
+        owner: Option<Owner>,
+        recipient: Recipient,
+        amount: Amount,
+    ) -> Result<Option<RawOutgoingMessage<SystemMessage, Amount>>, SystemExecutionError> {
+        if owner.is_some() {
+            ensure!(
+                authenticated_signer == owner,
+                SystemExecutionError::UnauthenticatedTransferOwner
+            );
+        }
+        ensure!(
+            amount > Amount::ZERO,
+            SystemExecutionError::IncorrectTransferAmount
+        );
+        let balance = match &owner {
+            Some(owner) => self.balances.get_mut_or_default(owner).await?,
+            None => self.balance.get_mut(),
+        };
+        balance
+            .try_sub_assign(amount)
+            .map_err(|_| SystemExecutionError::InsufficientFunding { balance: *balance })?;
+        if let Recipient::Account(account) = recipient {
```

Comment:
It would be safer to use `match` here, without a `_` arm. So if we ever add another account type, we don't accidentally burn the tokens here.